app/assets/javascripts/discourse/app/static/wizard/components/fields/radio.gjs

```diff
@@ -0,0 +1,72 @@
+import Component from "@glimmer/component";
+import { hash } from "@ember/helper";
+import { on } from "@ember/modifier";
+import { action, set } from "@ember/object";
+import PluginOutlet from "discourse/components/plugin-outlet";
+import concatClass from "discourse/helpers/concat-class";
+import withEventValue from "discourse/helpers/with-event-value";
+import icon from "discourse-common/helpers/d-icon";
+
+export default class Radio extends Component {
+  constructor() {
+    super(...arguments);
+
+    this._setSelected();
+  }
+
+  get field() {
+    return this.args.field;
+  }
+
+  @action
+  selectionChanged(input) {
+    this.field.value = input;
+    this._setSelected();
+  }
+
+  _setSelected() {
+    for (let choice of this.field.choices) {
+      set(choice, "selected", this.field.value === choice.id);
+    }
+  }
+
+  <template>
+    <div class="wizard-container__radio-choices">
+      {{#each @field.choices as |c|}}
```

Comment:
this the confusion it brings, here you used the @ and not this this.field, better use @ and this.args everywhere